client.lua

```diff
@@ -89,11 +89,61 @@ local function hasHarness(items)
     harness = _harness
 end
 
+-- Weapon Stress [Haroki]
+
+local function IsWhitelistedWeaponStress(weapon)
+    if weapon then
+        for _, v in pairs(config.WhitelistedWeaponStress) do
+            if weapon == v then
+                return true
+            end
+        end
+    end
+    return false
+end
+
+local hasWeapon = false
+
+local function startWeaponStressThread()
+    CreateThread(function()
+        while hasWeapon do
+            local ped = cache.ped
+            if IsPedShooting(ped) then
+                if math.random() < config.StressChance then
+                    TriggerServerEvent('hud:server:GainStress', math.random(1, 3))
+                end
+            end
+
+            if not IsPedArmed(ped, 4) then
+                hasWeapon = false
+            end
+            Wait(0)
+        end
+    end)
+end
+
+AddEventHandler('ox_inventory:currentWeapon', function(currentWeapon)
+    if not currentWeapon then
+        hasWeapon = false
+        return
+    end
+
+    local curGun = currentWeapon.name
+
+    if not IsWhitelistedWeaponStress(curGun) then
+        hasWeapon = true
+        startWeaponStressThread(curGun)
+    end
+end)
+
+-- Weapon Stress end
+
 RegisterNetEvent("QBCore:Client:OnPlayerLoaded", function()
     Wait(2000)
     local hudSettings = GetResourceKvpString('hudSettings')
     if hudSettings then loadSettings(json.decode(hudSettings)) end
     PlayerData = QBCore.Functions.GetPlayerData()
+    if IsPedArmed(PlayerPedId(), 4) then hasWeapon = true startWeaponStressThread() end
```

Comment:
Is this supposed to address my request to fix the edge case when the script restarts?
If it is, then unfortunately this won't really work correctly, since this check will be run when the player loads, not when the script starts (this check doesn't belong in any event at all).

`IsPedArmed` should be replaced by using `exports.ox_inventory:getCurrentWeapon()` and checking whether the weapon exists.

Also you are missing a `IsWhitelistedWeaponStress()` check here (you can use the previously mentioned `getCurrentWeapon()` export for this), but adding the check will lead to code duplication. Perhaps, to avoid this, the `startWeaponStressThread()` function should accept a `weapon` parameter and run the check there.

Reply:
Sorry stupid me, will resolve the issue :)